features/dd-sdk-android-trace/src/test/kotlin/com/datadog/android/trace/OtelTracerBuilderProviderTest.kt

```diff
@@ -446,4 +420,145 @@ internal class OtelTracerBuilderProviderTest {
+        val expectedNormalizedSampleRate = sampleRate / 100.0
+        val tracer = testedOtelTracerProviderBuilder.setSampleRate(sampleRate).build()
+            .tracerBuilder(fakeInstrumentationName).build()
+
+        // When
+        val coreTracer: CoreTracer = tracer.getFieldValue("tracer")
+
+        // Then
+        val config: Config = coreTracer.getFieldValue("initialConfig")
+        assertThat(config.traceSampleRate).isCloseTo(expectedNormalizedSampleRate, offset(0.005))
+    }
+
+    @Test
+    fun `M use user-keep priority W buildSpan { provided keep sample rate }`() {
+        // Given
+        val tracer = testedOtelTracerProviderBuilder
+            .setSampleRate(100.0)
+            .build()
+            .tracerBuilder(fakeInstrumentationName)
+            .build()
+
+        // When
+        val span = tracer
+            .spanBuilder(fakeOperationName)
+            .startSpan()
+        span.end()
+        val spanContext = span.spanContext as OtelSpanContext
+        val ddSpanContext = spanContext.delegate as DDSpanContext
+
+        // Then
+        // make sure we are flushing the spans in order to set the sampling priority
+        tracer.flush()
+        val priority = ddSpanContext.samplingPriority
+        assertThat(priority).isEqualTo(2)
+    }
+
+    @Test
+    fun `M use user-not-keep priority W buildSpan { provide not keep sample rate }`() {
+        // Given
+        val tracer = testedOtelTracerProviderBuilder
+            .setSampleRate(0.0)
+            .build()
+            .tracerBuilder(fakeInstrumentationName)
+            .build()
+
+        // When
+        val span = tracer
+            .spanBuilder(fakeOperationName)
+            .startSpan()
+        span.end()
+        val spanContext = span.spanContext as OtelSpanContext
+        val ddSpanContext = spanContext.delegate as DDSpanContext
+
+        // Then
+        // make sure we are flushing the spans in order to set the sampling priority
+        tracer.flush()
+        val priority = ddSpanContext.samplingPriority
+        assertThat(priority).isEqualTo(-1)
+    }
+
+    @Test
+    fun `M use user-keep or user-not-keep priority W buildSpan { provided random sample rate }`(
+        @DoubleForgery(min = 0.0, max = 100.0) sampleRate: Double
+    ) {
+        // Given
+        val tracer = testedOtelTracerProviderBuilder
+            .setSampleRate(sampleRate)
+            .build()
+            .tracerBuilder(fakeInstrumentationName)
+            .build()
+
+        // When
+        val span = tracer
+            .spanBuilder(fakeOperationName)
+            .startSpan()
+        span.end()
+        val spanContext = span.spanContext as OtelSpanContext
+        val ddSpanContext = spanContext.delegate as DDSpanContext
+
+        // Then
+        // make sure we are flushing the spans in order to set the sampling priority
+        tracer.flush()
+        val priority = ddSpanContext.samplingPriority
+        assertThat(priority).isIn(-1, 2)
```

Comment:
> [!WARNING]
> 
> This test is not testing whether we are using the sample rate passed in. We just test that when we set a sample rate, the span will have priority `manual drop` or `manual keep`.

A better test would be to create 100 spans with the tracer, and then count the number of `manual keep`, and make sure it's _relatively close_ to the input sample rate.

Reply:
I am not trying to test the sampling mechanism here, I am just trying to test that if you specify a sample rate you will either get `-1` or `2` but the intention is not to test the sampling logic.